bigquery/tests/system.py

```diff
@@ -167,9 +170,9 @@ def test_list_datasets(self):
             'newest' + unique_resource_id(),
         ]
         for dataset_name in datasets_to_create:
-            dataset = Config.CLIENT.dataset(dataset_name)
-            retry_403(dataset.create)()
-            self.to_delete.append(dataset)
+            created_dataset = Config.CLIENT.dataset(dataset_name)
+            retry_403(created_dataset.create)()
+            self.to_delete.append(created_dataset)
```

Comment:
Filed the cleanup to its own PR: https://github.com/GoogleCloudPlatform/google-cloud-python/pull/3551